api/payments.js

```diff
@@ -428,6 +428,42 @@ function getAccountPayments(request, response, next) {
     transactions.getAccountTransactions(options, response, callback);
   };
 
+  function attachDate(transactions, callback) {
+    var ledgerRequests = {};
+    var ledgerData = {};
+
+    async.each(transactions, function(transaction, next) {
+      // Prevent duplicate ledger requests
+      if (ledgerRequests[transaction.ledger_index] instanceof ripple.Request) {
+        return next(null);
+      }
+
+      ledgerRequests[transaction.ledger_index] = remote.requestLedger({
+        ledger_index: transaction.ledger_index
+      }, function (err, data) {
+        if (err) {
+          return next(err);
+        }
+
+        ledgerData[transaction.ledger_index] = data.ledger;
+
+        next(null);
+      });
+    }, function (err) {
+      if (err) {
+        return callback(err);
+      }
+
+      _.each(transactions, function (transaction) {
```

Comment:
Maybe create inner functions instead of using anonymous functions here? It would be easier to parse visually. I had a hard time tracking where the commas and closing `}`